docs/deploying_clearml/enterprise_deploy/openshift.md

```diff
@@ -0,0 +1,346 @@
+
+Set `enabled: false` for all ingresses in your `values.yaml` file:
+
+```
+# values.yaml
+apiserver:
+  ingress:
+    enabled: false
+
+fileserver:
+  ingress:
+    enabled: false
+
+webserver:
+  ingress:
+    enabled: false
+
+```
+
+##### Step 2: Create the Route Objects
+
+Create a YAML file (e.g., `clearml-routes.yaml`) with the following definitions to configure Routes for the ClearML 
+services. This single file defines all three required routes.
+
+```
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: clearml-enterprise-apiserver
+  namespace: clearml
+spec:
+  host: api.clearml.<YOUR_DOMAIN>
+  path: /
+  to:
+    kind: Service
+    name: clearml-clearml-enterprise-apiserver
+    weight: 100
+  port:
+    targetPort: 8008
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+
+---
+
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: clearml-enterprise-fileserver
+  namespace: clearml
+  annotations:
+    # This translates the NGINX proxy-read-timeout and proxy-send-timeout.
+    haproxy.router.openshift.io/timeout: 600s
+spec:
+  host: files.clearml.127-0-0-1.nip.io
+  path: /
+  to:
+    kind: Service
+    name: clearml-clearml-enterprise-fileserver
+    weight: 100
+  port:
+    targetPort: 8081
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+
+---
+
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: clearml-enterprise-webserver
+  namespace: clearml
+spec:
+  host: app.clearml.127-0-0-1.nip.io
+  path: /
+  to:
+    kind: Service
+    name: clearml-clearml-enterprise-webserver
+    weight: 100
+  port:
+    targetPort: 8080
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+Apply the file to your cluster: 
+
+```
+oc apply -f clearml-routes.yaml`
+```
+
+### ClearML Application Gateway
+
+#### Option 1: Using Kubernetes Ingress
+
+Enable the Ingress for the Application Gateway with the following `values.yaml` snippet.
+
+```
+# values.yaml
+ingress:
+  enabled: true
+  className: ""
+  hostname: "ttr.clearml.<YOUR_DOMAIN>"
+  tlsSecretName: "" # Optionally provide a secret for TLS
+```
+
+#### Option 2: Using OpenShift Routes
+
+To use Routes, you need to disable the default Ingress creation in the Helm chart, and then create the Route objects manually.
+
+##### Step 1: Disable Ingress in Helm Chart
+
+Set `enabled: false` for all ingresses in your `values.yaml` file:
+
+```
+# values.yaml
+ingress:
+  enabled: false
+```
+
+##### Step 2: Create the Route Object
+
+The following Route definition uses a wildcard host, which securely exposes both the primary gateway URL and any
+subdomains it requires. Create a file named `clearml-enterprise-app-gateway-route.yaml`.
+
+```
+# clearml-enterprise-app-gateway-route.yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: clearml-enterprise-app-gateway
+  namespace: clearml-tenant-a
+spec:
+  host: '*.ttr.clearml.127-0-0-1.nip.io'
```

Comment:
@pollfly as above,
1. Explain this is an example to be replaced with the desired hostname.
2. "ttr" is an obsolete term. A better example is "*.appgw.127..."